### PR TITLE
feat(users): user schema changes and REST endpoints

### LINKED ISSUE #23: [Task] User Module Schema Update and Interfaces

### PR DESCRIPTION
**Pull request by maxcao13**
Fixes #23 

## Description of the change:
- Adds GET, POST, PUT, and DELETE endpoints to specific users
- Updates User schema

## How to manually test:

1. Start dev server by `npm start`
2. Try this workflow:

if using [httpie](https://httpie.io/)
### POST a new user
```bash
http POST localhost:3000/api/users firstName=John lastName=Man email=testing@github.com gender=male birthday=2000-01-01 phoneNumber=1234567890 profilePicture=temp bio="Looking for housing!"
```
### GET a certain user:
```bash
http localhost:3000/api/users/${{id_of_user}}
```
### PUT a certain user:
```bash
http PUT localhost:3000/api/users/${{id_of_user}} profilePicture=temp2 bio="This is my update bio!"
```
### DELETE a certain user:
```bash
http DELETE localhost:3000/api/users/${{id_of_user}}
```
using cURL:

```bash
curl --request POST --header 'Content-Type: application/json' --data '{"firstName":"John","lastName":"Man","email":"testing@github.com"}' localhost:3000/api/users
```

etc... etc...

### CODE REVIEW

File: backend/src/routes/users.js [hunk 1]
```diff
@@ -16,4 +16,22 @@ router.post('/', (req, res) => {
...
+});
+
+router.put('/:userId', (req, res) => {
+	User.findByIdAndUpdate(req.params.userId, req.body, { new: true }).then((user) => {
+		res.json(user);
+	});
+});
+
+router.delete('/:userId', (req, res) => {
+	User.findByIdAndRemove(req.params.userId).then((user) => {
```
Comment (dlalaj) [inline backend/src/routes/users.js:32]: This should eventually cascade to delete the `Preferences` whose `userID` field matches the query param `userID` that the FE passes to the URL. We can fix this once I merge the schema for it.

**Review by dlalaj — CHANGES_REQUESTED** [review]

File: backend/src/routes/users.js (hunk 1 above)
Comment (maxcao13) [reply backend/src/routes/users.js:32]: So, a user needs a list of listings under their profile as well then.

Comment (dlalaj) [reply backend/src/routes/users.js:32]: Im not sure if you mean on the FE or BE. We probably dont need the list under the user model on the BE. Currently a lister has a `listerID` which is an `ObjectID` of the user who created the listing. As such-when we need to get listings for a user, we can simply query like so:
```commandline
Listing.find({listerID: "userID"}) - userID comes from the FE
```

**Comment by dlalaj** [comment]
👍🏻 Good to merge

**Review by dlalaj — APPROVED** [review]

**Pull request merged**